### PR TITLE
Fix canDecrypt when reindexing decrypted messages

### PR DESCRIPTION
**Pull request by arj03**
Testing box2 for real now with private groups and ran into this bug.

I'll be pushing up a new test in ssb-db2-box2 that shows the problem. There is probably also a fix in jitdb needed for that test to pass.

### CODE REVIEW

**Comment by arj03** [comment]
Hmm, this could probably use bsb to find the index ;) One moment

**Comment by arj03** [comment]
Fixed the O(n) (read ohh no ;-))

File: indexes/private.js [hunk 1]
```diff
@@ -192,7 +192,14 @@ module.exports = function (dir, sbot, config) {
       const content = tryDecryptContent(ciphertext, recBuffer, pValue)
       if (!content) return record
 
-      canDecrypt.push(recOffset)
+      if (!streaming) {
+        // since we use bsb for canDecrypt we need to ensure recOffset
+        // is inserted at the correct place when reindexing
+        var insertLocation = bsb.gt(canDecrypt, recOffset)
```
Comment (staltz) [inline indexes/private.js:198]:
```suggestion
        var insertLocation = bsb.gt(canDecrypt, recOffset)
```

**Review by staltz — APPROVED** [review]
Looks fine! Just a simple suggestion for a `var` change, but that's it. Good job on spotting the bsb

**Comment by arj03** [comment]
Thanks :)

**Pull request merged**